Std/Logic.lean

```diff
@@ -812,3 +812,9 @@ theorem Bool.eq_false_iff {b : Bool} : b = false ↔ b ≠ true :=
 theorem Bool.eq_iff_iff {a b : Bool} : a = b ↔ (a ↔ b) := by cases b <;> simp
 
 theorem ne_comm {α} {a b : α} : a ≠ b ↔ b ≠ a := ⟨Ne.symm, Ne.symm⟩
+
+@[simp] theorem Bool.beq_true (b : Bool) : (b == true) = b := by cases b <;> rfl
+
+@[simp] theorem Bool.beq_false (b : Bool) : (b == false) = !b := by cases b <;> rfl
+
+theorem Bool.not_beq_not (a b : Bool) : (!a == !b) = (a == b) := by cases a <;> cases b <;> rfl
```

Comment:
I had been trying out lemmas like these [here](https://github.com/leanprover/std4/pull/279/files#diff-e19379310e5f14110f720dcbdf5ca1f46a3c8a04c5b958617e9e467690bcd1f6R723-R732). As you can see, out of the three I only marked `(a != a') = false ↔ a == a'` with `simp`, because the RHS seems clearly simpler than the double negation. On the other hand, `(a == a') = false ↔ a != a'` doesn't seem to always be desirable, i.e., there were contexts in which I preferred the RHS or the LHS depending on my local hypotheses as well as the interactions with other `simp` rules. In general I feel a bit lost when deciding these things, and maybe it would even be worthwhile to pull in some rewriting theory like a macro that would apply Knuth-Bendix to a set of `simp` lemmas.

Reply:
(This is relevant insofar, iirc, sometimes `!a == b` was preferrable, and other times `a != b` (which is definitionally `!(a == b)`), again depending on local hyps and global `simp` behaviour.)

Reply:
According to #183, is it appropriate to add these lemmas?
```lean
/-! ### beq -/

@[simp] theorem beq_false_left : ∀ (x : Bool), (false == x) = !x := by decide
@[simp] theorem beq_false_right : ∀ (x : Bool), (x == false) = !x := by decide
@[simp] theorem beq_true_left : ∀ (x : Bool), (true == x) = x := by decide
@[simp] theorem beq_true_right : ∀ (x : Bool), (x == true) = x := by decide
theorem beq_not_left : ∀ (x y : Bool), (!x == y) = !(x == y) := by decide
theorem beq_not_right : ∀ (x y : Bool), (x == !y) = !(x == y) := by decide
@[simp] theorem beq_not_not : ∀ (x y : Bool), (!x == !y) = (x == y) := by decide
theorem beq_eq_not_xor : ∀ (x y : Bool), (x == y) = !xor x y := by decide

/-! ### bne -/

@[simp] theorem bne_false_left : ∀ (x : Bool), (false != x) = x := by decide
@[simp] theorem bne_false_right : ∀ (x : Bool), (x != false) = x := by decide
@[simp] theorem bne_true_left : ∀ (x : Bool), (true != x) = !x := by decide
@[simp] theorem bne_true_right : ∀ (x : Bool), (x != true) = !x := by decide
theorem bne_not_left : ∀ (x y : Bool), (!x != y) = (x == y) := by decide
theorem bne_not_right : ∀ (x y : Bool), (x != !y) = (x == y) := by decide
@[simp] theorem bne_not_not : ∀ (x y : Bool), (!x != !y) = (x != y) := by decide
theorem bne_eq_xor : ∀ (x y : Bool), (x != y) = xor x y := by decide
```

Reply:
Yes, these work but use `alias` when there are "better known" synonyms. (The meaning of "better known" is always up for debate but ideally one version should be the main one and it should be the only one with `@[simp]`, the others should be aliases.)

Reply:
> Perhaps we should replace `not_beq_not` with simp rules that move the negation to the top? e.g., `!a == b = !(a == b)`. and `a == !b = !(a == b)`.

I think this would break simpNF confluence. It's a strange thing but it's better to think of `!a == b` as `xor a b`.

Reply:
@fgdorais I agree.  I think normalizing to `xor` would make sense here.